internal/advertisement-operator/controller.go

```diff
@@ -160,6 +162,20 @@ func (r *AdvertisementReconciler) Reconcile(req ctrl.Request) (ctrl.Result, erro
 func (r *AdvertisementReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&advtypes.Advertisement{}).
+		WithEventFilter(predicate.Funcs{
```

Comment:
The event filters, as the name suggests, should be used to filter the events that could trigger the reconcile logic of the operator. Here we are putting the operational logic of the operator inside a predicate function. A viable solution would be to check the `adv.ObjectMeta.DeletionTimestamp`, done inside the reconcile logic, to determine if the object is being deleted. The `.DeletionTimestamp` is meaningful only when the resource has at least a finalizer, in our case the `advertisement` resource meets the condition, hence no need to add a new finalizer.

Reply:
Ok, but they are handler passed to the internal reconciler watcher, it is not doing anything on the adv resource, but is only updating an internal operator counter (that it is an internal state of operator, another thing that we should change). In my opinion, putting it in the event filter makes the reconcile function cleaner.

But if according to you it's better to move in inside the reconcile it's not a problem

Reply:
My 2 cents: I also think that this code should be put in the reconcile function. It is a bit strange to see the eventFilter used to modify the internal state of the operator instead of real filtering.